src/modules/async/async_sleep.c

```diff
@@ -95,6 +118,32 @@ int async_init_timer_list(void)
 	return 0;
 }
 
+int async_init_ms_timer_list(void)
+{
+	_async_ms_list = (struct async_ms_list *)shm_malloc(
+			sizeof(struct async_ms_list));
+	if(_async_ms_list == NULL) {
+		LM_ERR("no more shm\n");
+		return -1;
+	}
+	memset(_async_ms_list, 0, sizeof(struct async_ms_list));
+	if(lock_init(&_async_ms_list->lock) == 0) {
+		LM_ERR("cannot init lock \n");
+		shm_free(_async_ms_list);
+		_async_ms_list = 0;
+		return -1;
+	}
+	return 0;
+}
+
+int async_destroy_ms_timer_list(void)
```

Comment:
It is not necessary to shm_free the allocated _async_ms_list as well here?

Reply:
Good catch. Fixing now.